### PR TITLE
WIP - Go mod & bump k8s.io

### PR DESCRIPTION
**Pull request by adambkaplan**

### CODE REVIEW

**Comment by adambkaplan** [comment]
/assign @soltysh 

Please let me know if this is more clear. FYI I bumped to 1.16.2 in this one to ensure we have the latest security-related changes.

File: make/doc.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package make
+
+import (
+	// alpha-build-machinery contains non-go code used in makefiles
+	_ "github.com/openshift/library-go/alpha-build-machinery"
```
Comment (soltysh) [inline make/doc.go]: That's not sufficient, you also need bindata, see https://github.com/openshift/cluster-kube-controller-manager-operator/blob/master/pkg/dependencymagnet/doc.go and similarly to the other PR please name the package `dependencymagnet`.

File: go.mod [hunk 2]
```diff
@@ -35,9 +35,9 @@ require (
 	github.com/natefinch/lumberjack v2.0.0+incompatible // indirect
 	github.com/onsi/ginkgo v1.10.2 // indirect
 	github.com/onsi/gomega v1.7.0 // indirect
-	github.com/openshift/api v0.0.0-20190822120857-58aab2885e38
+	github.com/openshift/api v3.9.1-0.20190927182313-d4a64ec2cbd8+incompatible
```
Comment (soltysh) [inline go.mod]: Why?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,72 @@
+module github.com/openshift/cluster-openshift-controller-manager-operator
```
Comment (soltysh) [inline go.mod]: I want to see initial go.mod this commit is too large and makes the review hard.

File: Makefile [hunk 4]
```diff
@@ -28,7 +27,9 @@ $(call build-image,ocp-cluster-openshift-controller-manager-operator,$(IMAGE_REG
 # $5 - output
 # It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
 # and also hooked into {update,verify}-generated for broader integration.
-$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+#
+# TODO: make this compatible with go mod
```
Comment (soltysh) [inline Makefile]: See my other comment about missing bindata dep.

File: Makefile [hunk 5]
```diff
@@ -59,4 +60,9 @@ verify-codegen-crds: crd-schema-gen
 verify-codegen: verify-codegen-crds
 .PHONY: verify-codegen
 
-verify: verify-codegen
+# TODO: fix crd-schema-gen to work with go modules
```
Comment (soltysh) [inline Makefile]: Same comment as with the other PR fix in openshift/release not to run it.

**Comment by adambkaplan** [comment]
/close

Splitting the switch and k8s bump in two. Replaced in part by #127

**Pull request closed without merging**